File: app/src/main/java/com/fastaccess/ui/adapter/CommitsAdapter.java
package com.fastaccess.ui.adapter;

import android.support.annotation.NonNull;
import android.view.ViewGroup;

import com.fastaccess.data.dao.model.Commit;
import com.fastaccess.ui.adapter.viewholder.CommitsViewHolder;
import com.fastaccess.ui.widgets.recyclerview.BaseRecyclerAdapter;
import com.fastaccess.ui.widgets.recyclerview.BaseViewHolder;

import java.util.List;

/**
 * Created by Kosh on 11 Nov 2016, 2:07 PM
 */

public class CommitsAdapter extends BaseRecyclerAdapter<Commit, CommitsViewHolder, BaseViewHolder.OnItemClickListener<Commit>> {

    public CommitsAdapter(@NonNull List<Commit> data) {
        super(data);
    }

    @Override protected CommitsViewHolder viewHolder(ViewGroup parent, int viewType) {
        return CommitsViewHolder.newInstance(parent, this);
    }

    @Override protected void onBindView(CommitsViewHolder holder, int position) {
        holder.bind(getItem(position));
    }
}

File: commits/src/main/java/com/fastaccess/fasthub/commit/adapter/CommitsAdapter.kt
package com.fastaccess.fasthub.commit.adapter

import android.view.ViewGroup
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import com.fastaccess.data.model.FullCommitModel

class CommitsAdapter(
    private val callback: (FullCommitModel) -> Unit
) : ListAdapter<FullCommitModel, CommitsViewHolder>(object : DiffUtil.ItemCallback<FullCommitModel?>() {
    override fun areItemsTheSame(oldItem: FullCommitModel, newItem: FullCommitModel): Boolean = oldItem.id == newItem.id
    override fun areContentsTheSame(oldItem: FullCommitModel, newItem: FullCommitModel): Boolean = oldItem == newItem
}) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CommitsViewHolder = CommitsViewHolder(parent).apply {
        itemView.setOnClickListener {
            getItem(adapterPosition)?.let(callback)
        }
    }

    override fun onBindViewHolder(holder: CommitsViewHolder, position: Int) = holder.bind(getItem(position))
}